addons/website_sale/tests/test_website_editor.py

```diff
@@ -116,79 +122,123 @@ def test_extra_images_with_new_variant(self):
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i3, i1, i2, i4, i5, i6])
+            self.assertEqual(self.product.image_1920, i3)
+
+    def test_resequence_image_left(self):
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
             images = self.product._get_images()
-            self.assertEqual(images[2].image_1920, data_target)
-            self.assertEqual(images[0].image_1920, data_source)
-            # Test moving one to the left
-            data_source = images[2].image_1920
-            data_target = images[1].image_1920
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'left',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i3, i2, i4, i5, i6])
+
+    def test_resequence_image_right(self):
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
             images = self.product._get_images()
-            self.assertEqual(images[2].image_1920, data_target)
-            self.assertEqual(images[1].image_1920, data_source)
-            # Test moving one to the right
-            data_source = images[2].image_1920
-            data_target = images[3].image_1920
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'right',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i2, i4, i3, i5, i6])
+
+    def test_resequence_image_last(self):
+        self._create_product_images()
+        with MockRequest(self.product.env, website=self.website):
             images = self.product._get_images()
-            self.assertEqual(images[2].image_1920, data_target)
-            self.assertEqual(images[3].image_1920, data_source)
-            # Test moving one to the last position
-            data_source = images[2].image_1920
-            data_target = images[-1].image_1920
+            i1, i2, i3, i4, i5, i6 = self._get_product_image_data()
             self.WebsiteSaleController.resequence_product_image(
                 images[2]._name,
                 images[2].id,
                 'last',
             )
+            self.env['product.image'].invalidate_model()
+            self.assertListEqual(self._get_product_image_data(), [i1, i2, i4, i5, i6, i3])
+
+    def test_resequence_image_first_to_last(self):
+        """
+        Moving an image from first to last position is an edge case.
+        This test ensures it won't be accidentally broken.
```

Comment:
Why? It's a valid flow and the button is small enough that you can simply misclick it :sweat_smile: 

For once I find this comment useless, especially the second sentence.

Reply:
(FYI, I slightly adjusted the comment)